packages/core/src/logger/apiLogger.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  HttpRequest,
+  HttpResponse,
+  ApiLoggerInterface,
+  LoggerInterface,
+  Level,
+} from '../coreInterfaces';
+import { LoggingOptions } from './loggerOptions';
+import { NullLogger } from './nullLogger';
+
+export class ApiLogger implements ApiLoggerInterface {
+  private readonly _logger: LoggerInterface;
+
+  constructor(loggingOp: LoggingOptions) {
+    this._logger = loggingOp.logger ?? new NullLogger();
+  }
+
+  public logRequest(coreRequest: HttpRequest): void {
+    this._logger.log(
```

Comment:
I'm not sure if our implementation for ApiLogger combined with the default ConsoleLogger results in a well-formatted message for the SDK user from which they could debug their API call issues. Please share some sample output in the PR thread so that we can review them. Ideally, a user should be able to fully understand what HTTP request and HTTP response was transferred.

Reply:
![image](https://github.com/apimatic/apimatic-js-runtime/assets/109739087/e5adceec-a207-41f8-a784-96abecc766c7)